croquemort/reports.py

```diff
@@ -94,3 +97,45 @@ def compute_report(urls, filters, excludes, querystring, with_links=False):
     }
     template = env.get_template('report.html')
     return Response(template.render(context), mimetype='text/html')
+
+
+def compute_csv(urls, filters, excludes):
+    """Generate a streamed CSV of all data, optionally filtered."""
+
+    def generate():
+        """A generator is required to stream the actual CSV."""
+        fake_file = StringIO()
+        w = csv.writer(fake_file)
+
+        # Write the header.
+        w.writerow(('URL', 'Status', 'Content-Type', 'Updated'))
+        yield fake_file.getvalue()
+        fake_file.seek(0)
+        fake_file.truncate(0)
+
+        for url_hash, data in urls:
+            # Filtering.
+            data = apply_filters(data, filters, excludes)
+            if not data:
+                continue
+
+            # Statuses.
+            status = data.get('status', '')
```

Comment:
Thanks (but I know how `dict.get()` works), I just saw the test after but I didn't saw the value is serialized into the CSV output and so using the empty string as default value makes sense now ;)